### ISSUE TITLE
Clicking on child row toggle button causes all open child rows to be remounted

### ISSUE THREAD

**Issue by vpillinger** [matfish2/vue-tables-2#272]
- Vue.js version: 2.4.4
- consumed using: (browserify/webpack/rollup): Laravel Mix (webpack)
- relevant code: 
Create a table with a child row component that has
` mounted() {
  console.log('child mounted);
}`
- steps for reproducing the issue:
1) Click on the "+" or "-" button.
2) All child rows are destroyed and recreated triggering 'child mounted' console log for every open child row.            

Issue: It is inefficient if the child row template contains something like another table. In my use case, each click can generate 4-5 database calls by remounting components that get data from a server.

**Comment by matfish2** [comment]
Are you using the most recent version? I was unable to reproduce this.

**Comment by vpillinger** [comment]
Sorry, I should have been more specific. All OPEN child rows are destroyed and remounted. Closed ones are not remounted. I am working on a JSFiddle to demonstrate this.

**Comment by matfish2** [comment]
No need for a fiddle. Managed to reproduce. Working on a solution

**Comment by matfish2** [comment]
I was able to pinpoint the problem more accurately. When toggling a child row only open child rows that come **after** it are being remounted. I suppose this stems from the way vue reactivity works, but not sure yet what is the way to prevent it. 

Another consideration I have regarding child rows is that sometimes the user wants to retain the state of the child row after it was closed and reopened. For this reason I thought maybe `v-show` (css toggling) would be a better alternative to `v-if`(remounting the component). The problem with this approach is obviously the cost of rendering all the child rows in the page whenever the state of the table changes.

I need to do some research. Maybe come out with some middleway or let the consumer choose depending on his needs.

**Comment by fjoalland** [comment]
Hello @matfish2 , did you resolve this problem? I have a table vue with child rows. Their child rows call an API so it's useless to remounting every time all the child.

I tried to hide and show the row child with CSS property for controlling layout. But the problem will persist when I have to create the child for the first time, it will remounting all the child rows that come after this one.

**Comment by matfish2** [comment]
I am trying to figure this one out. It seems like it has to do with the way Vue handles reactivity on arrays and objects. 
I have simplified the data structure by replacing the object of toggled rows with an array of open rows, but the problem still persists, namely, Vue is updating the DOM not only for the changed element, but also for all subsequent ones.
I saw that as of 2.2 a `:key` attribute is required when using `v-for` to make change detection more efficient. However, since I have used JSX to write the template, it is irrelevant (I am using a `map` method instead to build the rows)
